src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1543,9 +1543,15 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
       We give this treatment to .vcxproj by default since no .vcxproj can target more
       than one framework.
+
+      Likewise if the dependency is for build ordering instead of an assembly reference
+      (ReferenceOutputAssembly=false), skip the checks since we can't know what TF
+      the output would need to be compatible with.
    -->
    <ItemGroup>
-      <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.SkipGetTargetFrameworkProperties)' == '' and '%(Extension)' == '.vcxproj'">
+      <_MSBuildProjectReferenceExistent
+        Condition="'%(_MSBuildProjectReferenceExistent.SkipGetTargetFrameworkProperties)' == '' and
+                   ('%(Extension)' == '.vcxproj' or '%(ReferenceOutputAssembly)' == 'false'">
```

Comment:
I'm slightly worried about this change. Could there be any projects today that ReferenceOutputAssembly=false but somehow depend on the TF check (and weren't broken by 15.5's changes to it)? I can't think of any, and I think this change makes a lot of sense. But I do worry.

Reply:
They would have had to have taken a dependency on `_MSBuildProjectReferenceExistent` to have custom logic do something with dependencies since they set `ReferenceOutputAssembly` to `false`, and we can't guarantee anything if they did that.

I think its also missing a trailing `)` right?

```diff
-('%(Extension)' == '.vcxproj' or '%(ReferenceOutputAssembly)' == 'false'
+('%(Extension)' == '.vcxproj' or '%(ReferenceOutputAssembly)' == 'false')
```

Reply:
missing trailing `)`: yup!

They wouldn't have had to take a dependency on `_MSBuildProjectReferenceExistent`, though--they could have just had

```xml
<ProjectReference Include="..\some\other.csproj">
  <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
</ProjectReference>
```

And it would have flowed through to here.